### PR TITLE
fix indexes_as_range

### PR DESCRIPTION
**Pull request by burrbull**
Fixes https://github.com/rust-embedded/svdtools/pull/178#issuecomment-1801433808

### CODE REVIEW

**Comment by burrbull** [comment]
cc @Emilgardis

File: svd-rs/src/dimelement.rs [hunk 1]
```diff
@@ -170,7 +170,11 @@ impl DimElement {
     pub fn indexes_as_range(&self) -> Option<RangeInclusive<u32>> {
         let mut integers = Vec::with_capacity(self.dim as usize);
         for idx in self.indexes() {
-            integers.push(idx.parse::<u32>().ok()?);
+            let val = idx.parse::<u32>().ok()?;
+            if val.to_string() != idx {
+                return None;
+            }
```
Comment (Emilgardis) [inline svd-rs/src/dimelement.rs:176]: what exactly does this fix?

that `01` != `1`?

Comment (Emilgardis) [reply svd-rs/src/dimelement.rs:176]: to note, I think it's the wrong way to fix the issue but maybe it's not

Comment (burrbull) [reply svd-rs/src/dimelement.rs:176]: > what exactly does this fix?
> 
> that `01` != `1`?

Yes

Comment (Emilgardis) [reply svd-rs/src/dimelement.rs:176]: can we add a comment here to explain why we need the name to be an exact match with the integer representation

Comment (burrbull) [reply svd-rs/src/dimelement.rs:176]: could you add comment yourself, please? I have issues with internet, maybe several days

**Review by Emilgardis — APPROVED** [review]
feel free to merge if the comment looks good

**Comment by Emilgardis** [comment]
ci stuck on waiting for runner 
image

**Pull request merged**